src/lib/common/MapLibreMap.svelte

```diff
@@ -2,106 +2,32 @@
   import { getStyleSpecification } from "lib/maplibre";
   import {
     Map,
-    NavigationControl,
-    ScaleControl,
-    type LayerSpecification,
     type LngLatBoundsLike,
-    type SourceSpecification,
+    type StyleSpecification,
   } from "maplibre-gl";
-  import { onDestroy, onMount, setContext } from "svelte";
-  import "maplibre-gl/dist/maplibre-gl.css";
   import { map as mapStore } from "stores";
+  import { onMount } from "svelte";
+  import { MapLibre, NavigationControl, ScaleControl } from "svelte-maplibre";
 
   export let style: string;
-  export let startBounds: LngLatBoundsLike | null = null;
+  export let startBounds: LngLatBoundsLike | undefined = undefined;
+
+  let styleSpec: string | StyleSpecification | null = null;
 
   let map: Map;
-  let mapContainer: HTMLDivElement;
   let loaded = false;
 
-  // Before creating the map, check if there's a hash, because one will get set below
-  // TODO Supposed to use a phantom type, not a string, as the key
-  let setCamera = !window.location.hash;
-  setContext("setCamera", setCamera);
-
-  // Adapted from https://github.com/dimfeld/svelte-maplibre/pull/60. We'll get
-  // this for free after migrating to svelte-maplibre.
-  let lastStyleLayerIds: Array<string> | undefined = undefined;
-  let lastStyleSourceIds: Array<string> | undefined = undefined;
-  let layersToReAddAfterStyleChange: Array<LayerSpecification> | undefined =
-    undefined;
-  let sourcesToReAddAfterStyleChange:
-    | Record<string, SourceSpecification>
-    | undefined = undefined;
+  $: if (loaded) {
+    // TODO Get rid of mapStore entirely, eventually?
```

Comment:
It might not be possible in every case, but more or less we want to whittle away all the places directly touching the map